online-shop/src/app/state/reducers/login.reducer.ts

```diff
@@ -0,0 +1,36 @@
+import {UserCredentials} from "../../interfaces/UserCredentials";
+import {createReducer, on} from "@ngrx/store";
+import {login, loginFailure, loginSuccessful, verifyAdmin} from "../actions/login.actions";
+import {User} from "../../interfaces/User";
+
+export interface LoginState {
+  loggedInUser: User | undefined;
+  error: string | null;
```

Comment:
Also, I wouldn't really place the error on the state. You don't need it for more than showing it once to the user in most cases so you can just pass it as param to the failure action and you just show it there.